haskell/private/actions/compile.bzl

```diff
@@ -124,6 +124,15 @@ def _compilation_defaults(
     compile_flags += hs.toolchain.ghcopts
     compile_flags += user_compile_flags
 
+    if hs.toolchain.is_darwin:
+        # assume `otool` and `install_name_tool` are available at the same location as `ar`
+        ar_bindir = paths.dirname(cc.tools.ar)
+
+        compile_flags += [
+            "-pgmotool=" + paths.join(ar_bindir, "otool"),
+            "-pgminstall_name_tool=" + paths.join(ar_bindir, "install_name_tool"),
+        ]
```

Comment:
I suspect this may fail for users that try to use a hermetic bindist cc toolchain like grail's llvm toolchain. I'm fine with this for now as it is an improvement over the status quo. But, from what I understand so far, this may not cover all use-cases, yet.

Reply:
Yes, we probably will need a better solution to this in the future.